File: src/main/java/de/tectoast/emolga/commands/pokemon/TypeInfoCommand.java
package de.tectoast.emolga.commands.pokemon;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.jsolf.JSONObject;

import java.util.LinkedList;
import java.util.List;

public class TypeInfoCommand extends Command {

    public TypeInfoCommand() {
        super("typeinfo", "Zeigt dir Informationen über einen Typen an", CommandCategory.Pokemon);
        setArgumentTemplate(ArgumentManagerTemplate.builder()
                .addEngl("type", "Typ", "Der Typ", Translation.Type.TYPE)
                .setExample("!typeinfo Wasser")
                .build());
    }

    @Override
    public void process(GuildCommandEvent e) {
        Translation tt = e.getArguments().getTranslation("type");
        String type = tt.getTranslation();
        JSONObject json = getTypeJSON();
        List<String> effectiveAgainst = new LinkedList<>();
        List<String> resistedBy = new LinkedList<>();
        List<String> weakAgainst = new LinkedList<>();
        List<String> resisted = new LinkedList<>();
        json.keySet().forEach(str -> {
            int damageTaken = json.getJSONObject(str).getJSONObject("damageTaken").getInt(type);
            if (damageTaken > 0) {
                String t = ((Translation) Translation.Type.TYPE.validate(str, Translation.Language.GERMAN, "default")).getTranslation();
                if (damageTaken > 1) {
                    if (damageTaken == 3) resistedBy.add(t + " **(immun)**");
                    else resistedBy.add(t);
                } else {
                    effectiveAgainst.add(t);
                }
            }
        });

        JSONObject typejson = json.getJSONObject(type).getJSONObject("damageTaken");
        typejson.keySet().forEach(str -> {
            int damageTaken = typejson.getInt(str);
            if (damageTaken > 0) {
                Translation t = (Translation) Translation.Type.TYPE.validate(str, Translation.Language.GERMAN, "default");
                if (t != null) {
                    if (damageTaken > 1) {
                        if (damageTaken == 3) resisted.add(t.getTranslation() + " **(immun)**");
                        else resisted.add(t.getTranslation());
                    } else {
                        weakAgainst.add(t.getTranslation());
                    }
                }
            }
        });
        e.reply("**" + tt.getOtherLang() + "**\n\n" +
                "- effektiv gegen\n" + String.join("\n", effectiveAgainst) + "\n\n" +
                "- wird resistiert von\n" + String.join("\n", resistedBy) + "\n\n" +
                "- ist schwach gegen\n" + String.join("\n", weakAgainst) + "\n\n" +
                "- resistiert\n" + String.join("\n", resisted));
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/pokemon/TypeInfoCommand.kt
package de.tectoast.emolga.commands.pokemon

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import java.util.*

class TypeInfoCommand : Command("typeinfo", "Zeigt dir Informationen über einen Typen an", CommandCategory.Pokemon) {
    init {
        argumentTemplate = ArgumentManagerTemplate.builder()
            .addEngl("type", "Typ", "Der Typ", Translation.Type.TYPE)
            .setExample("!typeinfo Wasser")
            .build()
    }

    override suspend fun process(e: GuildCommandEvent) {
        val tt = e.arguments.getTranslation("type")
        val type = tt.translation
        val json = typeJSON
        val effectiveAgainst: MutableList<String> = LinkedList()
        val resistedBy: MutableList<String> = LinkedList()
        val weakAgainst: MutableList<String> = LinkedList()
        val resisted: MutableList<String> = LinkedList()
        json.keySet().forEach {
            val damageTaken = json.getJSONObject(it).getJSONObject("damageTaken").getInt(type)
            if (damageTaken > 0) {
                val t = getTypeGerName(it)
                if (damageTaken > 1) {
                    if (damageTaken == 3) resistedBy.add("$t **(immun)**") else resistedBy.add(t)
                } else {
                    effectiveAgainst.add(t)
                }
            }
        }
        val typejson = json.getJSONObject(type).getJSONObject("damageTaken")
        typejson.keySet().forEach {
            val damageTaken = typejson.getInt(it)
            if (damageTaken > 0) {
                val t = getTypeGerName(it)
                    if (damageTaken > 1) {
                        if (damageTaken == 3) resisted.add("$t **(immun)**") else resisted.add(t)
                    } else {
                        weakAgainst.add(t)
                    }

            }
        }
        e.reply(
            "**${tt.otherLang}**\n\n- effektiv gegen\n${
                effectiveAgainst.joinToString("\n")
            }\n\n- wird resistiert von\n${
                resistedBy.joinToString("\n")
            }\n\n- ist schwach gegen\n${
                weakAgainst.joinToString("\n")
            }\n\n- resistiert\n${resisted.joinToString("\n")}"
        )
    }
}